src/main/java/com/cegeka/academy/service/challenge/ChallengeServiceImp.java

```diff
@@ -43,4 +49,25 @@ public void saveChallenge(ChallengeDTO challengeDTO) {
 
         logger.info("Challenge with id: " + challengeRepository.save(challenge).getId() + " has been saved");
     }
+
+    @Override
+    public Set<ChallengeDTO> getChallengesByUserId(long id) throws NotFoundException {
+
+        List<UserChallenge> userChallengesList = userChallengeRepository.findAllByUserId(id);
+
+        LinkedHashSet userChallengesSet = userChallengesList.stream().map(userChallenge -> {
+            Challenge challenge = userChallenge.getChallenge();
+            return ChallengeMapper.convertChallengeToChallengeDTO(challenge);
```

Comment:
Best practice pentru stream recomanda sa nu scrieti in map/foreach mai mult de o linie de cod. Prin urmare, poti extrage aceasta procesare intr-o metoda, pe care sa o apelezi in map